src/psyclone/tests/test_files/dynamo0p3/testkern_ref_elem_out_mod.F90

```diff
@@ -0,0 +1,75 @@
+!   this software without specific prior written permission.
+!
+! THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+! AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+! IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+! DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+! FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+! DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+! SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+! CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+! OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+! OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+! -----------------------------------------------------------------------------
+! Authors: R. W. Ford and A. R. Porter, STFC Daresbury Lab
+
+module testkern_ref_elem_out_mod
+  use argument_mod
+  use kernel_mod
+  use constants_mod
+  type, extends(kernel_type) :: testkern_ref_elem_out_type
+     type(arg_type), dimension(5) :: meta_args =    &
+          (/ arg_type(gh_real, gh_read),     &
+             arg_type(gh_field,gh_write,w1), &
+             arg_type(gh_field,gh_read, w2), &
+             arg_type(gh_field,gh_read, w2), &
+             arg_type(gh_field,gh_read, w3)  &
+             /)
+     type(reference_element_data_type), dimension(3) ::                &
+          meta_reference_element =                                     &
+          (/ reference_element_data_type(outward_normals_to_horizontal_faces), &
+             reference_element_data_type(normals_to_vertical_faces),           &
+             reference_element_data_type(outward_normals_to_vertical_faces) /)
+     integer :: iterates_over = cells
+   contains
+     procedure, nopass :: code => testkern_ref_elem_out_code
+  end type testkern_ref_elem_out_type
+contains
+
+  subroutine testkern_ref_elem_out_code(                                     &
+                           nlayers, ascalar, fld1, fld2, fld3, fld4,         &
+                           ndf_w1, undf_w1, map_w1, ndf_w2, undf_w2, map_w2, &
+                           ndf_w3, undf_w3, map_w3,                          &
+                           nfaces_h, nfaces_v,                               &
+                           out_normals_h, normals_v, out_normals_v)
+    integer, intent(in) :: nlayers, nfaces_h, nfaces_v
+    real(kind=r_def), intent(in) :: ascalar
+    real(kind=r_def), dimension(:), intent(out) :: fld1
+    real(kind=r_def), dimension(:), intent(in) :: fld2, fld3, fld4
+    integer, intent(in) :: ndf_w1, undf_w1, ndf_w2, undf_w2, ndf_w3, undf_w3
+    integer, dimension(:), intent(in) :: map_w1, map_w2, map_w3
+    real(kind=r_def), dimension(3, nfaces_h), intent(in) :: out_normals_h
```

Comment:
I think it would be worth expanding the argument names here a bit to be more consistent with other example, e.g. `out_normals_horiz` and `out_normals_vert` instead of just `out_normals_h` and `out_normals_v`.